src/libcrun/signals.c

```diff
@@ -595,14 +595,20 @@ str2sig (const char *name)
     {
       long int value;
 
+      if (!isdigit(name[0]))
+        goto error;
+
       errno = 0;
       value = strtol (name, NULL, 10);
-      if (errno == 0)
-        return value;
+      if (errno != 0)
+        goto error;
 
-      errno = ENOENT;
-      return -1;
+      return value;
     }
 
   return s->value;
+
+error:
+  errno = ENOENT;
```

Comment:
Shouldn't this be EINVAL?

Reply:
I try to keep the old behavor.

It seems that this errno is not used at all.

I think that EINVAL is better, too.

Reply:
Please change it.